pxc-80-backup/recovery-pvc-joiner.sh

```diff
@@ -26,12 +26,37 @@ function check_ssl() {
     fi
 
     if [ -f "$CA" -a -f "$KEY" -a -f "$CERT" ]; then
-        SOCAT_OPTS="openssl-connect:${RESTORE_SRC_SERVICE}:3307,cert=${CERT},key=${KEY},cafile=${CA},verify=1,commonname='',retry=30"
+        SOCAT_OPTS="openssl-connect:${RESTORE_SRC_SERVICE}:3307,reuseaddr,cert=${CERT},key=${KEY},cafile=${CA},verify=1,commonname='',retry=30"
     fi
 }
 
+parse_sst_info() {
+   local source_path=$1
+   local group=$2
+   local var=$3
+   local reval=""
+
+   reval=$(my_print_defaults -c "$source_path" $group | awk -F= '{if ($1 ~ /_/) { gsub(/_/,"-",$1); print $1"="$2 } else { print $0 }}' | grep -- "--$var=" | cut -d= -f2- | tail -1)
+
+   # use default if we haven't found a value
+   if [[ -z $reval ]]; then
+      [[ -n $4 ]] && reval=$4
+   fi
+
+   echo $reval
+}
+
+check_ssl
 ping -c1 $RESTORE_SRC_SERVICE || :
 rm -rf /datadir/*
+
 socat -u "$SOCAT_OPTS" stdio | xbstream -x -C /datadir --parallel=$(grep -c processor /proc/cpuinfo)
 socat -u "$SOCAT_OPTS" stdio | xbstream -x -C /datadir --parallel=$(grep -c processor /proc/cpuinfo)
-xtrabackup --prepare --target-dir=/datadir ${XB_USE_MEMORY+--use-memory=$XB_USE_MEMORY}
+
+transition_key=$(parse_sst_info "/datadir/sst_info" sst transition-key "")
+if [[ -n $transition_key ]]; then
+    encrypt_prepare_options="--transition-key=\$transition_key"
+    xtrabackup ${XB_USE_MEMORY+--use-memory=$XB_USE_MEMORY} --prepare --binlog-info=ON $encrypt_prepare_options --rollback-prepared-trx --xtrabackup-plugin-dir=/usr/lib64/xtrabackup/plugin --target-dir=/datadir
+else
+    xtrabackup --prepare --target-dir=/datadir ${XB_USE_MEMORY+--use-memory=$XB_USE_MEMORY}
+fi
```

Comment:
is it works like this?
```suggestion
transition_key=$(parse_sst_info "/datadir/sst_info" sst transition-key "")
if [[ -n $transition_key ]]; then
    encrypt_prepare_options="--transition-key=\$transition_key"
fi
xtrabackup ${XB_USE_MEMORY+--use-memory=$XB_USE_MEMORY} --prepare --binlog-info=ON $encrypt_prepare_options --rollback-prepared-trx --xtrabackup-plugin-dir=/usr/lib64/xtrabackup/plugin --target-dir=/datadir
```

Reply:
Yes, why?

Reply:
Done https://github.com/percona/percona-docker/pull/180/commits/819a80308ef74a48b1cd757252150c63825785b9
Can't accept your suggestions because it conflict with previous commit